modules/libmodbus/3.1.10/patches/add_build_file.patch

```diff
@@ -0,0 +1,78 @@
+--- /dev/null
++++ a/BUILD.bazel
+@@ -0,0 +1,74 @@
++load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
++
++cc_library(
++    name = "modbus",
++    srcs = glob(["src/*.c"]) + [
++        ":generate_config_h",
++        ":generate_modbus_version_h",
++    ],
++    hdrs = glob(["src/*.h"]),
++    copts = ["-DHAVE_CONFIG_H"],
++    includes = ["src"],
++    include_prefix = "libmodbus",
++    visibility = ["//visibility:public"],
++)
++
++cc_binary(
++    name = "libmodbus.so.0",
++    srcs = [
++    ],
++    linkopts = [
++        "-Wl,-soname,libmodbus.so.0",
++        # Linker script to hide non-modbus symbols.
++        "-Wl,--version-script",
++        "$(location modbus-versionscript.lds)",
++    ],
++    linkshared = 1,
++    deps = [
++        ":modbus",
++        ":modbus-versionscript.lds",
++    ],
++    visibility = ["//visibility:public"],
++)
++
++cc_library(
++    name = "modbus_shared",
```

Comment:
This seems a bit risky as consumers need to ensure that no target transitively depends on both `modbus_shared` and `modbus` (or end up having duplicate symbols). 

If you use a `cc_shared_library` for this rather than a `cc_binary`, users would always depend on `modbus` and could optionally opt into dynamic linking by adding this target to the `dynamic_deps` of a top-level target, which makes for a safer API.